delta/sourcing/src/main/scala/ch/epfl/bluebrain/nexus/sourcing/projections/Message.scala

```diff
@@ -69,14 +72,22 @@ final case class DiscardedMessage(offset: Offset, persistenceId: String, sequenc
   *
   * @param value the value of the message
   */
-final case class SuccessMessage[A](offset: Offset, persistenceId: String, sequenceNr: Long, value: A)
-    extends Message[A] {
+final case class SuccessMessage[A](
+    offset: Offset,
+    persistenceId: String,
+    sequenceNr: Long,
+    value: A,
+    warnings: Vector[RunResult.Warning]
+) extends Message[A] {
 
   def discarded: DiscardedMessage = DiscardedMessage(offset, persistenceId, sequenceNr)
 
   def failed(throwable: Throwable): FailureMessage[A] =
     FailureMessage(offset, persistenceId, sequenceNr, value, throwable)
 
+  def withWarning(warning: RunResult.Warning): SuccessMessage[A] = copy(warnings = warnings :+ warning)
```

Comment:
`addWarning`? the with seems to me more a replacement than an addition to the vector